parsers/caffe/caffeParser/opParsers/parsePReLU.cpp

```diff
@@ -37,10 +37,26 @@ ILayer* parsePReLU(INetworkDefinition& network, const trtcaffe::LayerParameter&
     {
         return nullptr;
     }
-    int nWeights = channelShared ? 1 : inputDims.d[1]; // Caffe treats second input dimension as channels
-    Dims slopesDims{inputDims.nbDims, {1}, {DimensionType::kSPATIAL}};
-    slopesDims.d[1] = nWeights;
+    int nWeights = channelShared ? 1 : inputDims.d[0];
 
+    Dims slopesDims{inputDims.nbDims, {nWeights, 1, 1, 1, 1, 1, 1, 1},
+                    {DimensionType::kCHANNEL, DimensionType::kSPATIAL, DimensionType::kSPATIAL, DimensionType::kSPATIAL,
+                     DimensionType::kSPATIAL, DimensionType::kSPATIAL, DimensionType::kSPATIAL, DimensionType::kSPATIAL}};
+    for(int i = 0; i < inputDims.nbDims; i++) {
+        int d{0};
+        switch(inputDims.type[i]) {
+        case DimensionType::kSPATIAL:
```

Comment:
`DimensionType` is deprecated, so I don't think we should rely on it here.

Reply:
Existing version has 2 bugs:
1. does not initialise all dimensions here
`Dims slopesDims{inputDims.nbDims, {1}, {DimensionType::kSPATIAL}};`
2. does not work with dynamic reshape.

If it is not possible to use `DimensionType` what is the alternative?

Reply:
The `DimensionType` field has been removed in TRT 8. Maybe we should fix this in the latest branch instead?